tests/unit/server/api/v1/datasets/test_search_dataset_records.py

```diff
@@ -242,3 +244,49 @@ async def test_with_record_without_vector(self, async_client: AsyncClient, owner
         assert response.json() == {
             "detail": f"Record `{record_without_vector.id}` does not have a vector for vector settings `{vector_settings.name}`"
         }
+
+    async def test_with_invalid_filter(self, async_client: AsyncClient, owner_auth_header: dict):
+        dataset = await DatasetFactory.create()
+
+        response = await async_client.post(
+            self.url(dataset.id),
+            headers=owner_auth_header,
+            json={
+                "query": {
+                    "text": {"q": "text", "field": "non-existent"},
+                },
+                "filters": {
+                    "and": [
+                        {
+                            "type": "terms",
+                            "scope": {"entity": "response", "question": "non-existent"},
+                            "values": ["value-a"],
```

Comment:
Maybe makes sense to parameterize this as having a wrong example per scope type (response, suggestion, and metadata property)

Reply:
Okay. I saw the specific unit tests for the validator. Forget this